specification/botservice/resource-manager/Microsoft.BotService/stable/2021-03-01/botservice.json

```diff
@@ -0,0 +1,2640 @@
+  "produces": [
+    "application/json"
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/botServices/{resourceName}": {
+      "put": {
+        "tags": [
+          "Bot"
+        ],
+        "description": "Creates a Bot Service. Bot Service is a resource group wide resource type.",
+        "operationId": "Bots_Create",
+        "x-ms-examples": {
+          "Create Bot": {
+            "$ref": "./examples/CreateBot.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Bot"
+            },
+            "description": "The parameters to provide for the created bot."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "If resource is created successfully or already existed, the service should return 200 (OK).",
+            "schema": {
+              "$ref": "#/definitions/Bot"
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (Created). Execution to continue asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/Bot"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Bot"
+        ],
+        "description": "Updates a Bot Service",
+        "operationId": "Bots_Update",
+        "x-ms-examples": {
+          "Update Bot": {
+            "$ref": "./examples/UpdateBot.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/Bot"
```

Comment:
Can you PATCH all of these properties?

Reply:
no change from previous version of the API. 

In implementation, BotProperties.MsaAppId is not patchable. I need guidence on whether marking it as readonly will be considered as breaking change.

Reply:
Yes, it would be a breaking change, but it would be a *good* breaking change, in that the API right now is improperly specified.  

I would suggest leaving this as is and creating a separate PR just with this correction - this will be easier to get through the breaking change review (which generally allows changes to properly model the API).

Reply:
keeping as is for back compat for now. will follow up with fix in separate PR